neuralprophet/utils.py

```diff
@@ -133,6 +133,32 @@ def reg_func_events(config_events, config_country_holidays, model):
     return reg_events_loss
 
 
+def reg_func_covariates(config_covariates, model):
+    """
+    Regularization of lagged covariates to induce sparsity
+
+    Parameters
+    ----------
+        config_covariates : OrderedDict
```

Comment:
I thought this is a `@dataclass Covar`

Reply:
Missed the notifications on your comments here, sorry.

I assume yes, it is quite inconsistent across the project so far:
1. OrderedDict https://github.com/ourownstory/neural_prophet/blob/main/neuralprophet/time_net.py#L71
2. configure.Covar https://github.com/ourownstory/neural_prophet/blob/29ecfbf41fa0d3613ecc9dbc83d7fc8ff1fbf795/neuralprophet/time_dataset.py#L226

I've adjusted it here, yet we should make this consistent across the project.

Reply:
It is also initialized as `OrderedDict` in case no value was provided, https://github.com/ourownstory/neural_prophet/blob/29ecfbf41fa0d3613ecc9dbc83d7fc8ff1fbf795/neuralprophet/forecaster.py#L466-L467 🤔

Reply:
Seems like some docstrings were not updated to our switch to using `@dataclass class` instead of `OrderedDict`.
Could you update all the docstrings (seems like primarily in time_net.py) that still mention `OrderedDict` in a separate PR?

Reply:
Sure, created an issue for it.